integration/tests/cook/util.py

```diff
@@ -1199,9 +1208,17 @@ def user_current_usage(cook_url, headers=None, **kwargs):
     return session.get('%s/usage' % cook_url, params=kwargs, headers=headers)
 
 
-def query_queue(cook_url, **kwargs):
+def query_queue(cook_url, allow_redirects=True, **kwargs):
     """Get current jobs via the queue endpoint (admin-only)"""
-    return session.get(f'{cook_url}/queue', **kwargs)
+    response = session.get(f'{cook_url}/queue', allow_redirects=False, **kwargs)
+    if allow_redirects and response.is_redirect:
```

Comment:
We're handling redirects manually here because the `requests` library is stripping out auth (and not re-applying it for some reason) when you have a redirect to a new hostname, which is what happens when /queue 307-redirects to the master. I'll add a comment here to clarify.

Reply:
Pushed the comment